### ISSUE TITLE
Code compile issue: “not declared in this scope”

### ISSUE THREAD

**Issue by phyl7h** [pymo/ppk_bluetooth#2]
Hi Pymo — thanks so much for this amazing project. I have had a Palm V keyboard waiting for this moment for 20 years.

I am a complete novice with Arduino and C++, and am having trouble compiling the code.

When I run the compile and upload, I get errors around line 586, in function “esp_sleep_wakeup_cause_t SleepAndWaitForWakeup()”. Arduino IDE advises:

> error:’gpio_hold_en’ was not declared in this scope; did you mean ‘gpio_mode_t’?

It gives similar errors for ‘gpio_wakeup_enable’ and ‘gpio_hold_dis’ - they were “not declared in this scope.”

Grateful for any help you can offer.

**Comment by pymo** [comment]
Hi,
It looks like the esp32 dependency is not found. Did you install the esp32 board dependency as instructed in the tutorial?

**Comment by phyl7h** [comment]
Thanks so much for the help Pymo. I thought I had installed that dependency but must not have. I removed and reinstalled Arduino IDE and all dependencies and the code installed without issue. Off to the races!

**Comment by phyl7h** [comment]
Hi Pymo — wanted to let you know that the code worked perfectly, and share my build. 

I wanted to fit everything on the keyboard itself without the need for an adapter. So I got rid of the slide-out stand to make room for a 304050 lipo (3mm thick). I also used an ESP32 C3 Mini board (https://a.aliexpress.com/_mLrhPGO). After using a Dremel to grind-down the plastic Palm dock there was room to just epoxy the new board directly to it. I used magnet wire for the leads to save bulk. I’m very happy that it actually folds. 

Thanks again for the inspiration and the code. 

![IMG_0411](https://github.com/pymo/ppk_bluetooth/assets/158836066/83525547-7952-4c14-a67e-e503ad29bb6d)
![IMG_0412](https://github.com/pymo/ppk_bluetooth/assets/158836066/57e21ca2-02af-48f8-a713-6f2ea3d92f6a)
![IMG_0413](https://github.com/pymo/ppk_bluetooth/assets/158836066/c9d55e5a-a086-4345-bc4f-e82c3038908f)

**Comment by pymo** [comment]
Thanks for sharing! I didn't have the nerve to dremel the connector off, so I went with the adapter route.

Also I couldn't find a small enough board that has li-po charger integrated (I believe your board didn't either?), so I actually started designing my own PCB, not based on ESP32-C3 but based on WCH CH582F. That should be small enough and have all the features needed to charge li-po. Stay tuned to this project if you are interested!
